### PR TITLE
Environment variable consistency

### PR DESCRIPTION
**Pull request by mireaulf**
Getting an env var in powershell can be done multiple ways
image

### CODE REVIEW

File: shell/probe/install.ps1 [hunk 1]
```diff
@@ -10,8 +10,15 @@ param(
   [String]$endpointId=$env:computername
 )
 
+function FromEnv { param ([string]$envVar, [string]$default)
```
Comment (makk94) [inline shell/probe/install.ps1:13]: nit: move params to next line just to be consistent with other function headers

Comment (makk94) [reply shell/probe/install.ps1:13]: also a small nit: be consistent across the files wrt if FromEnv function comes before or after the TLS  setting

**Review by makk94 — APPROVED** [review]

File: shell/probe/install.ps1 [hunk 2]
```diff
@@ -11,7 +11,15 @@ param(
 )
 
 [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
-$PRELUDE_API=if ($Env:PRELUDE_API) { $Env:PRELUDE_API } else { "https://api.preludesecurity.com" }
+
+function FromEnv { param ([string]$envVar, [string]$default)
+    $envVal = [Environment]::GetEnvironmentVariable($envVar, "Process")
+    if($envVal) {return $envVal}
```
Comment (makk94) [inline shell/probe/install.ps1:17]: if it's only a process-scoped variable, should you also set the user-scoped variable of the same name so the installed probe will work?

Comment (makk94) [reply shell/probe/install.ps1:17]: or could update the cmd we're passing the scheduled task to set the process-scoped env variable before starting raindrop

Comment (mireaulf) [reply shell/probe/install.ps1:17]: The install script sets the PRELUDE_TOKEN at User-level assuming that the script is not modified by a user, I don't see how the PRELUDE_TOKEN at Process-level would override User-level

Comment (makk94) [reply shell/probe/install.ps1:17]: was thinking about the PRELUDE_API variable - just so that the probe you install is talking to the right environment

Comment (mireaulf) [reply shell/probe/install.ps1:17]: Ah good catch!

File: shell/probe/install.ps1 [hunk 3]
```diff
@@ -74,7 +82,8 @@ if(Test-Path -path $probePath -PathType Leaf) {
 LogMessage "Determining OS"
 $dos = "windows-" + $Env:PROCESSOR_ARCHITECTURE
 $token=RegisterEndpoint
-[System.Environment]::SetEnvironmentVariable("PRELUDE_TOKEN", $token, "User")
+[Environment]::SetEnvironmentVariable("PRELUDE_TOKEN", $token, "User")
+[Environment]::SetEnvironmentVariable("PRELUDE_API", $PRELUDE_API, "Process")
```
Comment (makk94) [inline shell/probe/install.ps1:86]: the ScheduledTask will start a different process I think, so you need to add this line to the scheduled task action I believe

**Review by privateducky — APPROVED** [review]

**Pull request merged**